src/store/pizzaSlice.ts

```diff
@@ -0,0 +1,88 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { PizzaOrder, InitialStateType, OrderType } from '@ts/types/types';
+import FiltersEnum from '@ts/enums/enums';
+import { PizzaState } from '@ts/interfaces/interfaces';
+import { WritableDraft } from 'immer/dist/internal';
+
+const pizzaSlice = createSlice({
+  name: 'pizzas',
+  initialState: {
+    filter: 'Все',
+    sort: 'по популярности',
+    isLoading: false,
+    error: null,
+    order: [] as PizzaOrder[],
+  } as InitialStateType & PizzaState,
+  reducers: {
+    onFilterChange(state, action: PayloadAction<FiltersEnum>) {
+      state.filter = action.payload;
+    },
+
+    addPizza(state, { payload }: PayloadAction<PizzaOrder>) {
+      const pizza: PizzaOrder = payload;
+
+      const isParamsDifferent = state.order.find((o: PizzaOrder) => {
+        return (
+          o.pizza.id === pizza.pizza.id &&
+          o.params.type === pizza.params.type &&
+          o.params.size === pizza.params.size
+        );
+      });
+
+      if (isParamsDifferent) {
+        state.order.forEach((e: PizzaOrder) => {
+          if (e.pizza.id === pizza.pizza.id) {
+            e.params.count += 1;
+          }
+        });
+      } else {
+        pizza.params.count = 1;
+        state.order = [...state.order, pizza] as WritableDraft<OrderType[]> &
+          WritableDraft<PizzaOrder[]>;
+      }
+    },
+
+    deletePizza(state, action: PayloadAction<PizzaOrder>) {
+      const pizza = action.payload;
+      const orderedPizza = state.order.find(
+        (o: PizzaOrder) => o.pizza.id === pizza.pizza.id
+      );
+      if (orderedPizza) {
+        state.order.splice(
+          state.order.findIndex(
+            (e: PizzaOrder) => e.pizza.id === pizza.pizza.id
+          ),
+          1
+        );
+      }
+    },
+
+    decrementPizza(state, action: PayloadAction<PizzaOrder>) {
```

Comment:
использовать один подход - крайне важно
```suggestion
    decrementPizza(state, { payload }: PayloadAction<PizzaOrder>) {
```